my-image-message-app-backend/routes/userRouter.js

```diff
@@ -6,33 +6,45 @@ const auth = require("../middleware/auth");
 
 router.post("/register", async (req, res) => {
   try {
-    let { username, password, passwordCheck, message, image } = req.body;
-
-    if (!username || !password || !passwordCheck) {
+    let {
+      username,
+      password,
+      passwordCheck,
+      message,
+      image,
+      lastLoggedIn,
+    } = req.body;
+
+    if (!username || !password || !passwordCheck || !message) {
       return res.status(400).json({ msg: "Not all fields have been entered" });
+    } else {
+      if (password !== passwordCheck) {
+        return res
+          .status(400)
+          .json({ msg: "Enter the same password twice for verification." });
+      } else {
+        const existingUser = await User.findOne({ username: username });
+        if (existingUser) {
+          return res
+            .status(400)
+            .json({ msg: "An account with this username already exists." });
+        } else {
```

Comment:
This code can be simplified by removing else keyword because you use return in IF statement.